src/coreclr/jit/flowgraph.cpp

```diff
@@ -709,7 +709,7 @@ bool Compiler::fgIsThrow(GenTree* tree)
     // TODO-CQ: there are a bunch of managed methods in System.ThrowHelper
     // that would be nice to recognize.
 
-    return false;
+    return tree->AsCall()->IsNoReturn();
```

Comment:
Does this last line ever return true, given that  by this point `tree` must be a `CT_HELPER` call?

You might want to relax the check up at the top of the method and refactor the rest to handle user calls properly. Or perhaps merge this logic and `IsNoReturn` together somehow (say make helper calls that throw be no return calls too).

Also the helper checks should be reworked to use the `HelperCallProperties` mechanism.